src/main/java/frc/robot/autonomus/TrenchAuto.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.autonomus;
+
+
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.commands.command_groups.AutoShoot;
+import frc.robot.commands.command_groups.CollectCell;
+import frc.robot.motion_profiling.AutoPath;
+import frc.robot.motion_profiling.FollowPath;
+import frc.robot.subsystems.drivetrain.RotateDrivetrain;
+
+/**
+ * This auto command shoots 3 balls and then goes to collect more cells from the trench run
+ */
+public class TrenchAuto extends SequentialCommandGroup {
+    public TrenchAuto(StartingPose startingPose) {
+        AutoPath autoPath = startingPose == StartingPose.kFacingPowerPort ?
+            AutoPath.FacingPowerPortToTrenchStart : AutoPath.InLineWithTrenchToTrenchStart;
+        addCommands(
+            new AutoShoot(true),
+            // We get the desired angle by the initial pose rotation in the path.
+            new RotateDrivetrain(() ->
+                autoPath.getPath().getTrajectory().getInitialPose().getRotation().getDegrees()),
+            new FollowPath(autoPath),
+            deadline(
+                new FollowPath(AutoPath.InTrench),
+                new CollectCell()
+            ),
+            new AutoShoot(true)
```

Comment:
You have to rotate to the power port before shooting, and you cant shoot under the trench, so rotate, get close to the power port near the start of the trench run and then **shoot!**
**ALSO** the last AutoShoot(true) should be: AutoShoot() without shoot limit of 3 cells (you can make AutoShoot gets the number of cells to shoot).
good luck!